p2p/src/sync/peer_v1.rs

```diff
@@ -616,48 +612,44 @@ where
         }
 
         // The first header must be connected to a known block (it can be in
-        // the chainstate, pending_headers or requested_blocks).
+        // the chainstate or requested_blocks).
         let first_header_prev_id = *headers
             .first()
             // This is OK because of the `headers.is_empty()` check above.
             .expect("Headers shouldn't be empty")
             .prev_block_id();
 
+        // Note: we require a peer to send headers starting from a block that we already have
+        // in our chainstate or from one that we've already requested from the peer.
+        // I.e. peers shouldn't track what block headers they've sent us already and use
+        // the last header (best_sent_block_header) as a starting point for future HeaderList
+        // updates.
+        // This restriction is needed to prevent malicious peers from flooding the node with
+        // headers, potentially exhausting the node's memory.
+        // The downside of this is that the peer may have to send the same headers multiple times.
+        // So, to avoid extra traffic, an honest peer should't send header updates when the node
```

Comment:
There are tons of other ways to flood the node with traffic. This is a standard we can never attain.

Reply:
I've reformulated the comment a bit